corehq/apps/app_manager/models.py

```diff
@@ -2096,11 +2096,17 @@ def rename_lang(self, old_lang, new_lang):
     def validate_detail_columns(self, columns):
         from corehq.apps.app_manager.suite_xml.const import FIELD_TYPE_LOCATION
         from corehq.apps.locations.util import parent_child
+        from corehq.apps.locations.fixtures import should_sync_hierarchical_fixture
+
         hierarchy = None
         for column in columns:
             if column.field_type == FIELD_TYPE_LOCATION:
-                hierarchy = hierarchy or parent_child(self.get_app().domain)
+                domain = self.get_app().domain
+                project = Domain.get_by_name(domain)
                 try:
+                    if not should_sync_hierarchical_fixture(project):
+                        raise LocationXpathValidationError(_('Location references not supported in Case List.'))
```

Comment:
@czue This would take care of discontinuing the feature to use location references in Case List. Any suggestions for better error message? Any alternative that could be suggested?